indico/modules/categories/controllers/display.py

```diff
@@ -97,3 +101,33 @@ def _process_stats(self, stats, root=False):
 
     def _count_users(self):
         return User.find(is_deleted=False, is_pending=False).count()
+
+
+def _serialize_category(category, with_path=False):
+    data = {
+        'id': category.id,
+        'title': category.title,
+        'is_protected': category.is_protected,
+        'category_count': category.deep_children_count,
+        'event_count': category.deep_events_count,
+        'can_access': category.can_access(session.user)
+    }
+    if with_path:
+        data['path'] = [{'id': c.id, 'title': c.title} for c in category.parent_chain_query]
+    return data
+
+
+class RHCategoryInfo(RHDisplayCategoryBase):
+    _category_query_options = undefer('deep_children_count'), undefer('deep_events_count')
+
+    def _process(self):
+        return jsonify_data(category=_serialize_category(self.category, with_path=True),
+                            subcategories=[_serialize_category(c) for c in self.category.children])
+
+
+class RHCategorySearch(RH):
+    def _process(self):
+        starts_with = '{}%'.format(escape_like(request.args['q']))
+        query = Category.query.filter(Category.title.ilike(starts_with)).order_by(Category.title).limit(10)
+        query = query.options(undefer('deep_children_count'), undefer('deep_events_count'))
```

Comment:
Actually, you might want `.order_by(db.func.lower(Category.title))` to sort case-insensitive